extensions/security/runtime/src/main/java/io/quarkus/security/runtime/SecurityIdentityAssociation.java

```diff
@@ -55,7 +55,14 @@ public void setIdentity(Uni<SecurityIdentity> identity) {
     public SecurityIdentity getIdentity() {
         if (identity == null) {
             if (deferredIdentity != null) {
-                identity = deferredIdentity.await().indefinitely();
+                if (BlockingOperationControl.isBlockingAllowed()) {
+                    identity = deferredIdentity.await().indefinitely();
+                } else {
+                    throw new RuntimeException("Cannot call getIdentity() from the IO thread when lazy authentication " +
```

Comment:
Hi Stuart @stuartwdouglas Can you please add a small subsection with an example to https://quarkus.io/guides/security-built-in-authentication#proactive-authentication, we've been trying to accumulate all the hints like this one in our security docs.

Reply:
Thanks @stuartwdouglas :-)